x/gamm/pool-models/stableswap/pool.go

```diff
@@ -442,7 +442,7 @@ func validateScalingFactors(scalingFactors []uint64, numAssets int) error {
 	}
 
 	for _, scalingFactor := range scalingFactors {
-		if scalingFactor == 0 || int64(scalingFactor) <= 0 {
+		if scalingFactor == 0 || scalingFactor >= (1<<(62-types.MaxPoolAssets)) {
```

Comment:
Note to reviewer: this update overshoots by a little bit (adds a 1 bit buffer per asset when we actually only need log(maxNumAssets)), but I think it's more important to eliminate the class of scaling factor-related bugs than to get an exact bound since we don't expect people to use up that many bits for scaling factors anyway